### DISCUSSION TITLE
Spare State of dRAID pool does not become AVAIL after pool resilver has finished

### DISCUSSION THREAD

**Discussion by drearycold** [openzfs/zfs#12910] [Q&A] [answered]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Gentoo
Distribution Version	| N/A
Kernel Version	| 5.10.76-gentoo-r1
Architecture	| x86_64
OpenZFS Version	| zfs-2.1.2-r1-gentoo zfs-kmod-2.1.2-r0-gentoo

### Describe the problem you're observing
I've built a dRAID pool with specs draid1:4d:6c:1s. There was some interim problems with one of the disks, so spare have taken its place.
Since I know the disk is OK, I called `zpool clear` to reset the error counts. After reboot and resilver, `zpool status` shows that the state of spares is `INUSE`, which to my understanding should be `AVAIL`? Is this the right state? Or do I need to do something else?
Thanks for any help. I'm quite new to ZFS maintenance.

### Describe how to reproduce the problem
Let one disk make some read/write error, clear the error, wait for resilver to complete.

### Include any warning/errors/backtraces from the system logs

```
# zpool status
  pool: tank
 state: ONLINE
  scan: resilvered 1.27T in 08:45:06 with 0 errors on Wed Dec 22 19:44:15 2021
config:

        NAME                          STATE     READ WRITE CKSUM
        tank                          ONLINE       0     0     0
          draid1:4d:6c:1s-0           ONLINE       0     0     0
            scsi-350014e0eae48b89d    ONLINE       0     0     0
            scsi-350014e2e603e658d    ONLINE       0     0     0
            scsi-350014e2e603eb8ad    ONLINE       0     0     0
            scsi-350014e2e611923f9    ONLINE       0     0     0
            spare-4                   ONLINE       0     0     0
              scsi-350014e2e61248b68  ONLINE       0     0     0
              draid1-0-0              ONLINE       0     0     0
            scsi-350014e6e598d9392    ONLINE       0     0     0
        spares
          draid1-0-0                  INUSE     currently in use

errors: No known data errors

```

**Comment by behlendorf** [comment, marked as answer]
Good question.  Because the virtual `draid1-0-0` spare vdev is still included in the pool config you should expect the status to be `INUSE`.

```
        NAME                          STATE     READ WRITE CKSUM
        tank                          ONLINE       0     0     0
          draid1:4d:6c:1s-0           ONLINE       0     0     0
            scsi-350014e0eae48b89d    ONLINE       0     0     0
            scsi-350014e2e603e658d    ONLINE       0     0     0
            scsi-350014e2e603eb8ad    ONLINE       0     0     0
            scsi-350014e2e611923f9    ONLINE       0     0     0
            spare-4                   ONLINE       0     0     0
              scsi-350014e2e61248b68  ONLINE       0     0     0
>>>           draid1-0-0              ONLINE       0     0     0
            scsi-350014e6e598d9392    ONLINE       0     0     0
        spares
          draid1-0-0                  INUSE     currently in use
```

In order to return the spare the the `AVAIL` state you will need to run `zpool detach tank draid1-0-0`.  This would normally happen automatically after the failed drive was replaced and it had finished rebuilding back.  In your case, it's left the pool in the safest state until an administrator can determine the best course of action (i.e. perhaps that drive really is on its last legs and needs to go).

**Comment by drearycold** [comment]
Thanks!
`zpool  detach` restored spare state to `AVAIL`.
